game/song.cc

```diff
@@ -7,67 +7,70 @@
+        if (json.count(name) > 0) return json.at(name).get<T>();
+        return {};
+}
+
+Song::Song(nlohmann::json const& song): dummyVocal(TrackName::LEAD_VOCAL), randomIdx(rand()) {
+	path = getJsonEntry<std::string>(song, "txtFileFolder").value_or("");
+	filename = getJsonEntry<std::string>(song, "txtFile").value_or("");
+	artist = getJsonEntry<std::string>(song, "artist").value_or("");
+	title = getJsonEntry<std::string>(song, "title").value_or("");
+	language = getJsonEntry<std::string>(song, "language").value_or("");
+	edition = getJsonEntry<std::string>(song, "edition").value_or("");
+	creator = getJsonEntry<std::string>(song, "creator").value_or("");
+	genre = getJsonEntry<std::string>(song, "genre").value_or("");
+	cover = getJsonEntry<std::string>(song, "cover").value_or("");
+	background = getJsonEntry<std::string>(song, "background").value_or("");
+	video = getJsonEntry<std::string>(song, "videoFile").value_or("");
+	midifilename = getJsonEntry<std::string>(song, "midiFile").value_or("");
+	videoGap = getJsonEntry<double>(song, "videoGap").value_or(0.0);
+	start = getJsonEntry<double>(song, "start").value_or(0.0);
+	preview_start = getJsonEntry<double>(song, "previewStart").value_or(0.0);
+	m_duration = getJsonEntry<double>(song, "duration").value_or(0.0);
+	music[TrackName::BGMUSIC] = getJsonEntry<std::string>(song, "songFile").value_or("");
+	music[TrackName::LEAD_VOCAL] = getJsonEntry<std::string>(song, "vocals").value_or("");
+	music[TrackName::PREVIEW] = getJsonEntry<std::string>(song, "preview").value_or("");
+	music[TrackName::GUITAR] = getJsonEntry<std::string>(song, "guitar").value_or("");
+	music[TrackName::BASS] = getJsonEntry<std::string>(song, "bass").value_or("");
+	music[TrackName::DRUMS] = getJsonEntry<std::string>(song, "drums").value_or("");
+	music[TrackName::KEYBOARD] = getJsonEntry<std::string>(song, "keyboard").value_or("");
+	music[TrackName::GUITAR_COOP] = getJsonEntry<std::string>(song, "guitarCoop").value_or("");
+	music[TrackName::GUITAR_RHYTHM] = getJsonEntry<std::string>(song, "guitarRhythm").value_or("");
 	loadStatus = Song::LoadStatus::HEADER;
-	
-	if (song.has_field("VocalTracks")) {
-		for (unsigned i = 0; i < song.at("VocalTracks").as_number().to_uint32(); i++) {
-			std::string track = "DummyTrack" + std::to_string(i);
-			insertVocalTrack(track, VocalTrack(track));
-		}
+
+	for (size_t i = 0; i < getJsonEntry<size_t>(song, "vocalTracks").value_or(0); i++) {
+		std::string track = "DummyTrack" + std::to_string(i);
+		insertVocalTrack(track, VocalTrack(track));
+	}
+
+	if (getJsonEntry<bool>(song, "keyboardTracks").value_or(false)) {
+		instrumentTracks.insert(make_pair(TrackName::KEYBOARD, InstrumentTrack(TrackName::KEYBOARD)));
 	}
-	
-	if (song.has_field("KeyboardTracks")) {
-			instrumentTracks.insert(make_pair(TrackName::KEYBOARD, InstrumentTrack(TrackName::KEYBOARD)));
+
+	if (getJsonEntry<bool>(song, "drumTracks").value_or(false)) {
```

Comment:
careful field was "DrumTracks" and now is "drumTracks"; Moreover, are you sure this is  boolean value? shouldn't it be :
`if (getJsonEntry<std::string>(song, "drumTracks"))` ?

Reply:
This was changed in a previous commit for the following reasons:
- use camlCase json names (compliant with Google recommandations)
- use boolean value instead of int for this field since we only encode a boolean.

Reply:
ok, my bad